libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Core.hs

```diff
@@ -98,36 +94,189 @@ class
       (NonEmpty (SpecRep (PredicateFailure (EraRule rule era))))
       (SpecRep (State (EraRule rule era)))
 
-conformsToImpl ::
-  forall (rule :: Symbol) fn era.
+  translateInputs ::
+    Environment (EraRule rule era) ->
+    State (EraRule rule era) ->
+    Signal (EraRule rule era) ->
+    ExecContext fn rule era ->
+    ImpTestM
+      era
+      ( SpecRep (Environment (EraRule rule era))
+      , SpecRep (State (EraRule rule era))
+      , SpecRep (Signal (EraRule rule era))
+      )
+  default translateInputs ::
+    ( ForAllRuleTypes (SpecTranslate (ExecContext fn rule era)) rule era
+    , ToExpr (SpecRep (Environment (EraRule rule era)))
+    , ToExpr (SpecRep (State (EraRule rule era)))
+    , ToExpr (SpecRep (Signal (EraRule rule era)))
+    ) =>
+    Environment (EraRule rule era) ->
+    State (EraRule rule era) ->
+    Signal (EraRule rule era) ->
+    ExecContext fn rule era ->
+    ImpTestM
+      era
+      ( SpecRep (Environment (EraRule rule era))
+      , SpecRep (State (EraRule rule era))
+      , SpecRep (Signal (EraRule rule era))
+      )
+  translateInputs env st sig ctx = do
+    agdaEnv <- expectRight . runSpecTransM ctx $ toSpecRep env
+    logEntry $ "agdaEnv:\n" <> showExpr agdaEnv
+    agdaSt <- expectRight . runSpecTransM ctx $ toSpecRep st
+    logEntry $ "agdaSt:\n" <> showExpr agdaSt
+    agdaSig <- expectRight . runSpecTransM ctx $ toSpecRep sig
+    logEntry $ "agdaSig:\n" <> showExpr agdaSig
+    pure (agdaEnv, agdaSt, agdaSig)
+
+  testConformance ::
+    ( ShelleyEraImp era
+    , SpecTranslate (ExecContext fn rule era) (State (EraRule rule era))
+    , NFData (SpecRep (Environment (EraRule rule era)))
+    , NFData (SpecRep (State (EraRule rule era)))
+    , NFData (SpecRep (Signal (EraRule rule era)))
+    , ToExpr (SpecRep (Environment (EraRule rule era)))
+    , ToExpr (SpecRep (State (EraRule rule era)))
+    , ToExpr (SpecRep (Signal (EraRule rule era)))
+    ) =>
+    Environment (EraRule rule era) ->
+    State (EraRule rule era) ->
+    Signal (EraRule rule era) ->
+    Property
+  testConformance = defaultTestConformance @fn @era @rule
+
+checkConformance ::
+  ( ToExpr (TestRep (PredicateFailure (EraRule rule era)))
+  , ToExpr (TestRep (State (EraRule rule era)))
+  , Eq (TestRep (PredicateFailure (EraRule rule era)))
+  , Eq (TestRep (State (EraRule rule era)))
+  ) =>
+  Either
+    (NonEmpty (TestRep (PredicateFailure (EraRule rule era))))
+    (TestRep (State (EraRule rule era))) ->
+  Either
+    (NonEmpty (TestRep (PredicateFailure (EraRule rule era))))
+    (TestRep (State (EraRule rule era))) ->
+  ImpTestM era ()
+checkConformance implResTest agdaResTest = do
+  let
+    failMsg =
+      unlines
+        [ ""
+        , "===== DIFF ====="
+        , diffExpr implResTest agdaResTest
+        , ""
+        , "Legend:"
+        , "\t\ESC[91mImplemenation"
```

Comment:
You have a small typo here, just in case it annoys you later on, otherwise we can fix it next time

Reply:
Thanks, fixed it.